core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -858,19 +844,33 @@ object SparkSubmit extends CommandLineUtils {
     require(path != null, "path cannot be null.")
     val uri = Utils.resolveURI(path)
     uri.getScheme match {
-      case "file" | "local" =>
-        path
-
+      case "file" | "local" => path
+      case "https" | "http" | "ftp" => path
       case _ =>
         val fs = FileSystem.get(uri, hadoopConf)
-        val tmpFile = new File(Files.createTempDirectory("tmp").toFile, uri.getPath)
+        val tmpFile = new File(Files.createTempDirectory("tmp").toFile, new Path(uri).getName)
         // scalastyle:off println
         printStream.println(s"Downloading ${uri.toString} to ${tmpFile.getAbsolutePath}.")
         // scalastyle:on println
         fs.copyToLocalFile(new Path(uri), new Path(tmpFile.getAbsolutePath))
         Utils.resolveURI(tmpFile.getAbsolutePath).toString
     }
   }
+
+  private def resolveGlobPaths(paths: String, hadoopConf: HadoopConfiguration): String = {
+    require(paths != null, "paths cannot be null.")
+    paths.split(",").filter(_.trim.nonEmpty).flatMap { path =>
```

Comment:
minor, but you should actually trim the strings, not just for filtering purposes.